presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -182,13 +240,40 @@ public static Slice split(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType
     {
         checkCondition(limit > 0, INVALID_FUNCTION_ARGUMENT, "Limit must be positive");
         checkCondition(limit <= Integer.MAX_VALUE, INVALID_FUNCTION_ARGUMENT, "Limit is too large");
-        List<String> result = Splitter.on(delimiter.toStringUtf8())
-                .limit((int) limit)
-                .splitToList(string.toStringUtf8());
-        return toStackRepresentation(result, VARCHAR);
+        checkCondition(!UnicodeUtil.isEmpty(delimiter), INVALID_FUNCTION_ARGUMENT, "The delimiter may not be the empty string");
+
+        final List<Slice> parts = new ArrayList<>();
+
+        int index = 0;
+        while (index < string.length()) {
+            final int splitIndex = UnicodeUtil.findUtf8IndexOfString(string, index, string.length(), delimiter);
+            //
+            // Found split?
+            if (splitIndex < 0) {
+                break;
+            }
+            //
+            // Non empty split?
+            if (splitIndex > index) {
+                if (parts.size() == limit - 1) {
```

Comment:
I have changed the logic so that the code now is backward-compatible (processing of empty splits)
`SPLIT('a..b', '.', 3)` now produces `'a', '', 'b'`